app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/AlertPlayer.java

```diff
@@ -534,23 +534,25 @@ private void VibrateNotifyMakeNoise(Context context, AlertType alert, String bgV
                 .setDeleteIntent(snoozeIntent(context, minsFromStartPlaying));
 
         if (profile != ALERT_PROFILE_VIBRATE_ONLY && profile != ALERT_PROFILE_SILENT) {
-            if (minsFromStartPlaying >= MAX_VIBRATING_MINUTES) {
-                // Before this, we only vibrate...
-                float volumeFrac = (float) (minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
-                volumeFrac = Math.min(volumeFrac, 1);
-                if (profile == ALERT_PROFILE_MEDIUM) {
-                    volumeFrac = (float) 0.7;
-                }
-                Log.d(TAG, "VibrateNotifyMakeNoise volumeFrac = " + volumeFrac);
-                boolean overrideSilent = alert.override_silent_mode;
-                boolean forceSpeaker = alert.force_speaker;
-                if (notSilencedDueToCall()) {
-                    if (overrideSilent || isLoudPhone(context)) {
-                        playFile(context, alert.mp3_file, volumeFrac, forceSpeaker, overrideSilent);
-                    }
-                } else {
-                    Log.i(TAG, "Silenced Alert Noise due to ongoing call");
+            float volumeFrac = (float) (minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
+            // While minsFromStartPlaying <= MAX_VIBRATING_MINUTES, we only vibrate ...
+            volumeFrac = Math.max(volumeFrac, 0); // Limit volumeFrac to values greater than and equal to 0
+            volumeFrac = Math.min(volumeFrac, 1); // Limit volumeFrac to values less than and equal to 1
+            if (!Pref.getBoolean("delay_ascending_3min", true) && volumeFrac < 0.3) {
+                volumeFrac = (float) 0.3; // If delay_ascending_3min is disabled, we never only vibrate.
+            }
+            if (profile == ALERT_PROFILE_MEDIUM) {
```

Comment:
This condition check should be moved before line 537 and the remaining `volumeFrac` calculation in the else part, because line 545 is a constant assignment.

Reply:
Please explain what you mean by this.

Reply:
There is no need to set the `volumeFrac` before the condition `profile == ALERT_PROFILE_MEDIUM` evaluates to `true`. That's why I suggest to change it to:

```java
            float volumeFrac;
            if (profile == ALERT_PROFILE_MEDIUM) {
                volumeFrac = (float) 0.7;
            } else
                volumeFrac = (float) (minsFromStartPlaying - MAX_VIBRATING_MINUTES) / (MAX_ASCENDING_MINUTES - MAX_VIBRATING_MINUTES);
                // While minsFromStartPlaying <= MAX_VIBRATING_MINUTES, we only vibrate ...
                volumeFrac = Math.max(volumeFrac, 0); // Limit volumeFrac to values greater than and equal to 0
                volumeFrac = Math.min(volumeFrac, 1); // Limit volumeFrac to values less than and equal to 1
                if (!Pref.getBoolean("delay_ascending_3min", true) && volumeFrac < 0.3) {
                    volumeFrac = (float) 0.3; // If delay_ascending_3min is disabled, we never only vibrate.
                }
            }
```

Reply:
We have to ask the original developer who wrote this.
If I was to speculate, I would say the following:
If the ascending volume profile is selected, the volume is updated every minute.

But, if the medium volume profile is selected, any modifications made by the ascending volume conditional need to be overwritten by 70%.  Otherwise, the ascending volume scheme will overwrite the medium volume.

By placing 

if (profile == ALERT_PROFILE_MEDIUM) {
                    volumeFrac = (float) 0.7;

after the entire ascending volume code, if medium volume is selected, it will overwrite.

What I don't understand is how High volume overwrites it if selected.

But, I have not changed that.  So, it should still work.

I am now testing Medium setting on my device.
After, I will test High.

Reply:
> But, if the medium volume profile is selected, any modifications made by the ascending volume conditional need to be overwritten by 70%.

Thats true. `volumeFrac = (float) 0.7` is an assignment of a constant and is completely independent of the ascending volume profile. That's why it is not necessary to do the ascending calculations if the medium profile is selected.
Hence, I don't think it is necessary to ask the original developer because my suggestion does not change the logic. BTW: I've simplified it further which is more readable.

Reply:
Thanks for the suggestions.
I would like to leave it.  
This PR changes something that has been in the code for longer than 6 years.
I'd like to minimize the number of lines being changed to increase the chance of it being merged.  I do that by only changing what is necessary.
Changing the Medium volume assignment, which seems to be redundant, is not necessary to accomplish the objective of the PR, to avoid delaying the ascending volume alerts.

There could be another PR that would delete the medium volume assignment here, and perform tests to confirm functionality remains unchanged.  I believe if medium or high volume profiles are selected, they are assigned somewhere else in the code.
There are many lines in the code that can be removed.
But, proposing to change something that makes no difference to functionality only jeopardizes its approval.

A PR that is not merged, no matter how correct and complete it is, is useless.